### ISSUE TITLE
Changing barcode on a part manually - either a bug or I'm doing it wrong..

### ISSUE THREAD

**Issue by babogan** [ArctosDB/arctos#8364]
**Describe the bug**
Our paleo curator has moved many specimens and thus the locations need updated. In the "edit" screen on individual entries, I am unable to edit the current barcode location (for example https://arctos.database.museum/guid/ALMNH:Paleo:20461 is currently DMRCMHB1073 and it needs to be changed to DMRCMHB238)

**To Reproduce**
1. Go to (https://arctos.database.museum/guid/ALMNH:Paleo:20461)
2. Click on "edit parts"
3. Type in 'add to barcode' field: DMRCMHB238
4. See error: 
 ERROR_ID	5306DFAC-40B2-4CED-92F4FB1A65D5B5C4
ERROR_TYPE	SQL
ERROR_MESSAGE	ERROR: permission denied for table container
ERROR_DETAIL	
ERROR_SQL	update container set parent_container_id=21543496 where container_id=23818195

It seems it can't add the item to two locations... but shouldn't the new location I'm inputting overwrite the old one? There is no way that I am seeing where I can delete the old location if this is the case, except maybe through https://arctos.database.museum/findContainer.cfm ? But I think that is a bit out of the way when I'm updating a few quickly as things are moved. 

This is something I used to be able to do and I have had students do in the past and it has worked in the past!

**Comment by dustymc** [comment]
That looks like you just don't have update container access, which should come from @aklompma.

**Comment by babogan** [comment]
Hi Dusty!
@aklompma came by and he says he doesn't have this access either so he couldn't grant it to me! We both need this access.
@jcabbott13 I think we need this access granted to me for all collections to proceed with the object tracking project next semester - could you give your blessing here?

**Comment by campmlc** [comment]
@babogan If you are just moving an object from one barcoded location(container) to another, you don't want to change the barcode in the barcode field of Edit Container. To use Edit Container for this purpose, you use the Move to Barcode at the lower left. The alternative, if you are using scannable barcodes, is just to scan the barcodes into their new containers, or, if you don't have scannable barcodes, you can use Container:Move or Container: BulklMove in the Tools: Containers Menu. I highly recommend scannable barcodes.

**Comment by campmlc** [comment]
Just FYI - you should only ever try to change the actual barcode of a container for situations where the original barcode has fallen off or become illegible. For moving items between containers, use the various move container tools. I'd be happy to do a zoom over this to help clarify.

**Comment by babogan** [comment]
This isn't a container edit or barcode change to be clear, this is simply moving the object from one barcoded drawer to another!
I more meant that we needed to change the associated container by use of the barcodes

**Comment by campmlc** [comment]
Yes, you should be able to do that with update container. But again, don't change the assigned barcode next to the label field in Edit Container for this.

**Comment by babogan** [comment]
Got it! Thanks

**Comment by campmlc** [comment]
Available tools to move an object from one parent container to another: 
https://arctos.database.museum/moveContainer.cfm
https://arctos.database.museum/loaders/move_container.cfm
Edit Container: 
![Screenshot 2024-12-12 09 41 27](https://github.com/user-attachments/assets/de631149-79d5-4deb-81be-5119bca8a087)

Or ideally, use positions and scan items into the position space.
![Screenshot 2024-12-12 09 44 32](https://github.com/user-attachments/assets/23edd3ad-ceb8-41c1-a7b5-abea6a442973)

**Comment by dustymc** [comment]
@aklompma you're listed as an admin for the collection so I think I can safely give you whatever rights you want (but @jcabbott13 please let me know if I'm overstepping).

Please review the documentation:

* https://handbook.arctosdb.org/documentation/container.html

and you have update_container - you may need to log out/in to see it. (I did not give you admin or manage - https://arctos.database.museum/Admin/user_roles.cfm - let me know if you want them.)

**Comment by babogan** [comment]
@dustymc thank you, much appreciated. We've been discussing this with @jcabbott13 and he wants me to be the global admin for all of the collections, I believe, as he doesn't check into ARCTOS much. @jcabbott72 (is this his newer account by chance?)

**Comment by dustymc** [comment]
> simplify this for single entries

@babogan just do what you were trying to do (but let us know if it starts getting tedious, there are lots of bulk tools available).

>admin for all of the collections

Happy to help, probably best done in an email to me and @mkoo

**Comment by campmlc** [comment]
@babogan yes, you can use the part editor as well. But you would just replace the old barcode associated with the part with the new one. Container IDs and barcodes are different things -the simplest tools available use the barcodes to move objects. Again, happy to do a zoom to clarify. These are tools I use daily, and there are different tools for different situations that make things more efficient and simpler.

**Comment by babogan** [comment]
I have access to the bulk tools and have successfully used them prior! I'm hoping to create a workflow for single/small changes that students will be able to be trained on quickly an will be able to apply as they go. Thanks so much for helping with this!

**Comment by babogan** [comment]
@campmlc Thanks so much! Would love to have a little meeting to clarify. We're planning to work on object tracking quite a bit in the new year

**Comment by campmlc** [comment]
And apologies, I read this incorrectly initially as using Edit Container vs Part Edit - that makes a huge difference!

**Comment by babogan** [comment]
No problem! I definitely explained it in a confusing manner!

**Comment by babogan** [comment]
@campmlc thanks so much!! would help a ton especially as I start trying to train others. We don't like to train bulkloading unless students are going to be available for more than one semester so knowing alternatives is always a huge help. We come back the first week of January but may be busy. If you're available the second or third week I have open availability except on Mondays!

**Comment by campmlc** [comment]
Yes, perhaps the third week? Let's check in the second week and we can set a time.

**Comment by babogan** [comment]
Thanks! I'll get the plan on my calendar. Really appreciate it.

**Comment by campmlc** [comment]
sounds good. This issue is still pending approval for your global admin access - so please follow up with @dustymc and @mkoo for that.

**Comment by babogan** [comment]
We've moved to email with that issue - thanks so much!

**Comment by mkoo** [comment]
Just logged in to say that this is being addressed via email!
